renderer/nyaovim-app.ts

```diff
@@ -164,6 +164,117 @@ Polymer({
+                // get current vim mode
+                var m = client.commandOutput('echo mode()');
+                m.then(value => {
+                    //  mode() returns a strange '\n' at the beginning, why?
+                    value = value.trim();
+                    if (value.length > 0) {
+                        const ch = value[0];
+                        if (ch == 'v'  // visual mode
+                            || ch == 'V' // visual line mode
+                            || (ch == '^' && value[1] == 'V') // visual block mode
+                           ) {
+                            const command = '"+y';
+                            client.input(command);
+                        } else {
+                            // execute the default copy command
+                            const webContents = ThisBrowserWindow.webContents;
+                            webContents.copy();
+                        }
+                    }
+                });
+            });
+
+            ipc.on('nyaovim:select-all', (_: Electron.IpcRendererEvent) => {
+                // get current vim mode
+                var m = client.commandOutput('echo mode()');
+                m.then(value => {
+                    //  mode() returns a strange '\n' at the beginning, why?
+                    value = value.trim();
+                    if (value.length > 0) {
+                        const ch = value[0];
+
+                        if (ch == 'n') {
+                            const command = 'ggVG';
+                            client.input(command);
+                        } else {
+                            // switch to normal mode first
+                            const command = '<esc>ggVG';
+                            client.input(command);
+                        }
+                    }
+                });
+            });
+
+            ipc.on('nyaovim:cut', (_: Electron.IpcRendererEvent) => {
+                // get current vim mode
+                var m = client.commandOutput('echo mode()');
+                m.then(value => {
+                    //  mode() returns a strange '\n' at the beginning, why?
+                    value = value.trim();
+                    if (value.length > 0) {
+                        const ch = value[0];
+
+                        if (ch == 'v'  // visual mode
+                            || ch == 'V' // visual line mode
+                            || (ch == '^' && value[1] == 'V') // visual block mode
+                           ) {
+                            const command = '"+x';
+                            client.input(command);
+                        } else {
+                            // other modes
+                            const webContents = ThisBrowserWindow.webContents;
+
+                            // execute the default cut command
+                            webContents.cut();
+                        }
```

Comment:
I think below is simpler rather than this `if` statement.

```javascript
const command = ch === 'n' ? 'ggVG' : '<Esc>ggVG';
client.input(command);
```